python/lsst/ts/ess/common/abstract_command_handler.py

```diff
@@ -0,0 +1,215 @@
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["AbstractCommandHandler"]
+
+from abc import ABC, abstractmethod
+import logging
+import typing
+
+import jsonschema
+
+from .command_error import CommandError
+from .constants import Command, Key, ResponseCode
+from .schema import CONFIG_JSCHEMA
+
+
+class AbstractCommandHandler(ABC):
+    """Handle incoming commands and send replies. Apply configuration and read
+    sensor data.
+
+    Parameters
+    ----------
+    callback: `Callable`
+        The callback coroutine handling the sensor telemetry. This can be a
+        coroutine that sends the data via a socket connection or a coroutine in
+        a test class to verify that the command has been handled correctly.
+    simulation_mode: `int`
+        Indicating if a simulation mode (> 0) or not (0) is active.
+
+    The commands that can be handled are:
+
+        configure: Load the configuration that is passed on with the command
+        and connect to the devices specified in that configuration. This
+        command can be sent multiple times before a start is received and only
+        the last configuration is kept.
+        start: Start reading the sensor data of the connected devices and send
+        it as plain text via the socket. If no configuration was sent then the
+        start command is ignored. Once started no configuration changes can be
+        done anymore.
+        stop: Stop sending sensor data and disconnect from all devices. Once
+        stopped, configuration changes can be done again and/or reading of
+        sensor data can be started again.
+
+    """
+
+    valid_simulation_modes = (0, 1)
+
+    @abstractmethod
+    def __init__(self, callback: typing.Callable, simulation_mode: int) -> None:
+        self.log = logging.getLogger(type(self).__name__)
+        if simulation_mode not in self.valid_simulation_modes:
+            raise ValueError(
+                f"simulation_mode={simulation_mode} "
+                f"not in valid_simulation_modes={self.valid_simulation_modes}"
+            )
+
+        self.simulation_mode = simulation_mode
+
+        self._callback = callback
+        self._configuration: typing.Optional[typing.Dict[str, typing.Any]] = None
+        self._started = False
+
+        self.dispatch_dict: typing.Dict[str, typing.Callable] = {
```

Comment:
Done

Reply:
I think the issue is that you allow any kwargs for these callables. If so, I think the best you can do, based on [this discussion](https://github.com/python/mypy/issues/5876) rather than the mypy docs is:
```
typing.Callable[..., <return type>]
```

Reply:
I will try to solve this in ticket DM-31414.